nucliadb/nucliadb/search/search/query.py

```diff
@@ -188,3 +188,22 @@ async def paragraph_query_to_pb(
     ResourceProcessingStatus.PROCESSED: Resource.ResourceStatus.PROCESSED,
     ResourceProcessingStatus.ERROR: Resource.ResourceStatus.ERROR,
 }
+
+
+def pre_process_query(user_query: str) -> str:
```

Comment:
I think it might be a good idea to use a Strategy design pattern here to have more flexibility switching between preprocessing algorithms

Reply:
I agree that if the business logic becomes more complex, a Strategy pattern would help. However, I don't see the need right now. If it's OK, I'll leave a comment in the code.